srml/contract/src/rent.rs

```diff
@@ -54,14 +55,16 @@ fn try_evict_or_and_pay_rent<T: Trait>(
 		Some(ContractInfo::Alive(contract)) => contract,
 	};
 
-	// Calculate an effective block number, i.e. after adjusting for handicap.
-	let effective_block_number = <system::Module<T>>::block_number().saturating_sub(handicap);
-
 	// How much block has passed since the last deduction for the contract.
-	let blocks_passed = match effective_block_number.saturating_sub(contract.deduct_block) {
-		// Rent has already been paid
-		n if n.is_zero() => return RentOutcome::Exempted,
-		n => n,
+	let blocks_passed = {
+		// Calculate an effective block number, i.e. after adjusting for handicap.
+		let effective_block_number = <system::Module<T>>::block_number().saturating_sub(handicap);
+
+		match effective_block_number.saturating_sub(contract.deduct_block) {
+			// Rent has already been paid
+			n if n.is_zero() => return RentOutcome::Exempted,
```

Comment:
why no just a if instead of match if

Reply:
that's a good question! Match allows you to return `n` and to match against `0`. Typically, but in this case I had to use this awkward `n if n.is_zero()` so no advantage whatsoever